handlers/films.go

```diff
@@ -85,18 +91,48 @@ func (gh ghibliHandler) GetFilms(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		log.Printf("Failed to fetch films: %s", err)
 		w.WriteHeader(http.StatusInternalServerError)
-		w.Write([]byte(err.Error()))
+		w.Header().Set("Content-Type", "application/json")
+		response := map[string]interface{}{"response": err.Error()}
+		json.NewEncoder(w).Encode(response)
 		return
 	}
 
-	jsonFilms, err := json.Marshal(films)
+	w.WriteHeader(http.StatusOK)
+	w.Header().Set("Content-Type", "application/json")
+	response := map[string]interface{}{"response": films}
+	json.NewEncoder(w).Encode(response)
+	return
+}
+
+func (gh ghibliHandler) TestWorkers(w http.ResponseWriter, r *http.Request) {
+	var jobs []*workerspool.Job
+
+	for i := 1; i <= 50; i++ {
+		jobs = append(jobs, workerspool.NewJob(func(d interface{}) []string {
+			fmt.Println("Executing job:", d.(int))
+			time.Sleep(100 * time.Millisecond)
+			return nil
+		}, i, 5))
+	}
+
+	pool := workerspool.NewWorkersPool(jobs, 25)
+	pool.Run()
```

Comment:
retrieve data from pool by using pool.Results, but also please fix the close of channels, as this is making your threads to wait indefinitely to write over closed channels, if there are doubts pleas read again the material on how waitgroups and channels work. cause TestWorkers and GetFilmsConcurrently suffer from a lot of deadlocks